From e571b5f143d251143210edfda7e8e3b85d91a30f Mon Sep 17 00:00:00 2001
From: Martin Nolte <nolte@mathematik.uni-freiburg.de>
Date: Mon, 10 Jul 2017 18:31:02 +0200
Subject: [PATCH] [bugfix] remove documentation install targets unless Doxygen
 / LaTeX found

Currently, the documentation install targets fail if Doxgen or LaTeX are
not present. This patch removes the corresponding install targets in
these cases.
---
 cmake/modules/DuneDoc.cmake     | 22 +++++++++++----------
 cmake/modules/DuneDoxygen.cmake | 42 ++++++++++++++++++++---------------------
 2 files changed, 33 insertions(+), 31 deletions(-)

diff --git a/cmake/modules/DuneDoc.cmake b/cmake/modules/DuneDoc.cmake
index 3977b57b..9436d636 100644
--- a/cmake/modules/DuneDoc.cmake
+++ b/cmake/modules/DuneDoc.cmake
@@ -102,16 +102,18 @@ include(DuneSphinxCMakeDoc)
 include(DuneDoxygen)
 
 macro(create_doc_install filename targetdir dependency)
-  dune_module_path(MODULE dune-common RESULT scriptdir SCRIPT_DIR)
-  get_filename_component(targetfile ${filename} NAME)
-  set(install_command ${CMAKE_COMMAND} -D FILES=${filename} -D DIR=${CMAKE_INSTALL_PREFIX}/${targetdir} -P ${scriptdir}/InstallFile.cmake)
-  # create a custom target for the installation
-  add_custom_target(install_${targetfile} ${install_command}
-    COMMENT "Installing ${filename} to ${targetdir}"
-    DEPENDS ${dependency})
-  # When installing, call cmake install with the above install target and add the file to install_manifest.txt
-  install(CODE "execute_process(COMMAND \"${CMAKE_COMMAND}\" --build \"${CMAKE_BINARY_DIR}\" --target install_${targetfile} )
-            LIST(APPEND CMAKE_INSTALL_MANIFEST_FILES ${CMAKE_INSTALL_PREFIX}/${targetdir}/${targetfile})")
+  if(LATEX_USABLE)
+    dune_module_path(MODULE dune-common RESULT scriptdir SCRIPT_DIR)
+    get_filename_component(targetfile ${filename} NAME)
+    set(install_command ${CMAKE_COMMAND} -D FILES=${filename} -D DIR=${CMAKE_INSTALL_PREFIX}/${targetdir} -P ${scriptdir}/InstallFile.cmake)
+    # create a custom target for the installation
+    add_custom_target(install_${targetfile} ${install_command}
+      COMMENT "Installing ${filename} to ${targetdir}"
+      DEPENDS ${dependency})
+    # When installing, call cmake install with the above install target and add the file to install_manifest.txt
+    install(CODE "execute_process(COMMAND \"${CMAKE_COMMAND}\" --build \"${CMAKE_BINARY_DIR}\" --target install_${targetfile} )
+              LIST(APPEND CMAKE_INSTALL_MANIFEST_FILES ${CMAKE_INSTALL_PREFIX}/${targetdir}/${targetfile})")
+  endif()
 endmacro(create_doc_install)
 
 
diff --git a/cmake/modules/DuneDoxygen.cmake b/cmake/modules/DuneDoxygen.cmake
index 26f86b82..a3a2b003 100644
--- a/cmake/modules/DuneDoxygen.cmake
+++ b/cmake/modules/DuneDoxygen.cmake
@@ -94,26 +94,26 @@ macro(add_doxygen_target)
     add_custom_target(doxygen_${DOXYGEN_TARGET}
       DEPENDS ${DOXYGEN_OUTPUT})
     add_dependencies(doc doxygen_${DOXYGEN_TARGET})
-  endif()
 
-  # Use a cmake call to install the doxygen documentation and create a
-  # target for it
-  include(GNUInstallDirs)
-  # When installing call cmake install with the above install target
-  install(CODE
-    "execute_process(COMMAND ${CMAKE_COMMAND} --build ${CMAKE_BINARY_DIR} --target doxygen_${ProjectName}
-        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
-      file(GLOB doxygenfiles
-        GLOB ${CMAKE_CURRENT_BINARY_DIR}/html/*.html
-        ${CMAKE_CURRENT_BINARY_DIR}/html/*.js
-        ${CMAKE_CURRENT_BINARY_DIR}/html/*.png
-        ${CMAKE_CURRENT_BINARY_DIR}/html/*.css
-        ${CMAKE_CURRENT_BINARY_DIR}/html/*.gif)
-      set(doxygenfiles \"\${doxygenfiles}\")
-      foreach(_file \${doxygenfiles})
-         get_filename_component(_basename \${_file} NAME)
-         LIST(APPEND CMAKE_INSTALL_MANIFEST_FILES ${CMAKE_INSTALL_FULL_DOCDIR}/doxygen/\${_basename})
-       endforeach()
-       file(INSTALL \${doxygenfiles} DESTINATION ${CMAKE_INSTALL_FULL_DOCDIR}/doxygen)
-       message(STATUS \"Installed doxygen into ${CMAKE_INSTALL_FULL_DOCDIR}/doxygen\")")
+    # Use a cmake call to install the doxygen documentation and create a
+    # target for it
+    include(GNUInstallDirs)
+    # When installing call cmake install with the above install target
+    install(CODE
+      "execute_process(COMMAND ${CMAKE_COMMAND} --build ${CMAKE_BINARY_DIR} --target doxygen_${ProjectName}
+          WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+        file(GLOB doxygenfiles
+          GLOB ${CMAKE_CURRENT_BINARY_DIR}/html/*.html
+          ${CMAKE_CURRENT_BINARY_DIR}/html/*.js
+          ${CMAKE_CURRENT_BINARY_DIR}/html/*.png
+          ${CMAKE_CURRENT_BINARY_DIR}/html/*.css
+          ${CMAKE_CURRENT_BINARY_DIR}/html/*.gif)
+        set(doxygenfiles \"\${doxygenfiles}\")
+        foreach(_file \${doxygenfiles})
+           get_filename_component(_basename \${_file} NAME)
+           LIST(APPEND CMAKE_INSTALL_MANIFEST_FILES ${CMAKE_INSTALL_FULL_DOCDIR}/doxygen/\${_basename})
+         endforeach()
+         file(INSTALL \${doxygenfiles} DESTINATION ${CMAKE_INSTALL_FULL_DOCDIR}/doxygen)
+         message(STATUS \"Installed doxygen into ${CMAKE_INSTALL_FULL_DOCDIR}/doxygen\")")
+  endif()
 endmacro(add_doxygen_target)
-- 
2.13.0

